### DISCUSSION TITLE
[Pinocchio 3.0.0] What is the ddq_dq calculated by computeConstraintDynamicsDerivatives function for unit quaternions?

### DISCUSSION THREAD

**Discussion by zixinz990** [stack-of-tasks/pinocchio#2261] [Q&A] [answered]
For a floating joint, q is a 7-dim vector, while dq and ddq are both 6-dim vectors. I noticed that, the ddq_dq calculate by computeConstraintDynamicsDerivatives function is a 6-by-6 matrix. I was wondering if the attitude-related part in dq_ddq is the exact derivative for unit quaternion, or you transform it to Euler angles or other 3-params representation implicitly.

**Comment by jcarpent** [comment, marked as answer]
For unit quaternions, the $\partial q$ is not directly related to the parametrization of the orientation but corresponds to a small increment on the tangent space of rotations, meaning $R(q+\partial q) = R(q)\text{exp}(\partial q)$
